src/odb/src/lefout/lefout.cpp

```diff
@@ -144,7 +158,34 @@ void lefout::writeObstructions(dbBlock* db_block)
   dbBox* block_bounding_box = db_block->getBBox();
   for (dbTechLayer* tech_layer : obstruction_layers) {
     fprintf(_out, "    LAYER %s ;\n", tech_layer->getName().c_str());
-    writeBox("  ", block_bounding_box);
+
+    if (pin_clearance) {
+      Polygon90Set polygon_set;
+
+      for (dbBTerm* term : db_block->getBTerms()) {
+        for (dbBPin* pin : term->getBPins()) {
+          for (dbBox* box : pin->getBoxes()) {
+            if (box->getTechLayer() == tech_layer) {
+              polygon_set.insert(Rectangle(box->xMin(), box->yMin(), box->xMax(), box->yMax()));
+            }
+          }
+        }
+      }
+
+      Polygon90Set bloat_set = Polygon90Set(polygon_set + tech_layer->getSpacing());
```

Comment:
Min space is not necessarily correct.  A more than min-width pin shape would become a violation.

Reply:
@maliberty any thoughts on how best to handle this? Do we have to also look at the width of the obstruction next to the pin and lookup the spacing table based on the two widths, or can we get away with just using the widest clearance all the time?

I would guess obstructions between pins are almost always going to be big enough to invoke the widest spacing rule.

I'm also not sure if this direction (of creating clearance around pins) is what all the existing tools do and expect.

Reply:
> I would guess obstructions between pins are almost always going to be big enough to invoke the widest spacing rule.
I think that is ok but you need to make sure there is actually no shape closer than what you are cutting away.

> I'm also not sure if this direction (of creating clearance around pins) is what all the existing tools do and expect.
I've seen it both ways - either with a blockage over the pins or a cutaway around them.  If the pins are not on the edge you must cut away the blockage but on the edge it isn't really a big difference.